### PR TITLE
Improve completion logic and move it into python. Add zsh support.

### PR DESCRIPTION
**Pull request by mpcusack-color**
Having more of the logic it in python makes having support for zsh and bash much easier. 

The completion logic now looks at all current arguments and complete the first missing required one until all required arguments are given. Assumes you will be using exclusively named arguments even for required args. This is different than how people have been using `clr` (wasn't supported until recent refactor). Positional arguments are still fully supported in clr, but the shell completion will encourage people to use named ones.

Example:
```
def cmd_argtest(self, a, b, c=4, d=None, e=False, f=True):
```
```
$ clr help argtest
usage: clr system:argtest [--a A] [--b B] [--c C] [--d D] [--e | --noe]
                          [--f | --nof]
                          [a] [b] [c] [d]
...
```

Using shell completion. Bold is completed for you.
<pre>
clr sy[TAB]<b>stem:</b>a[TAB]<b>rgtest</b> [TAB]<b>--a </b>AAA [TAB]<b>--b </b>BBB [TAB]<b>--</b>[TAB][TAB]
<b>--c     --d     --e     --noe   --f     --nof </b>   
<b>clr system:argtest --a AAA --b  BBB --</b>e [TAB][TAB]
<b>--c     --d     --f     --nof   </b>
</pre>

### CODE REVIEW

File: clr/commands.py [hunk 1]
```diff
@@ -494,7 +499,96 @@ def cmd_completion_arg(self, command_name, partial="", bools_only=False):
...
+            if param.kind == param.VAR_POSITIONAL:
+                has_var_positional = True
+                continue
+
+            required = param.default == Signature.empty
+            missing_args = missing_required_args if required else missing_optional_args
+            arg_names = [f"--{param.name}"]
+
+            if type(param.default) == bool:
+                arg_names.append(f"--no{param.name}")
```
Comment (abatilo) [inline clr/commands.py:563]: For my clarity, is this going to make params `--noparam` or `--no-param`? Seems like it'll be the prior. Any objections to doing the `--no-` prefix?

Comment (mpcusack-color) [reply clr/commands.py:563]: `--noparam`

Would also prefer `--no-param` if we were starting from scratch, but this is done to match the old parsing scheme.

Comment (mpcusack-color) [reply clr/commands.py:563]: Also, this PR is just the shell completion and doesn't actually define the flags.

**Comment by mpcusack-color** [comment]
> Seems reasonable as far as I can tell? I'm honestly not very familiar with how you configure shell completion.

I've tackled bash completion a couple times before. zsh is new to me. Both are messy.

Next (hopefully last for a while) step is to update the environment setup scripts in `color/color` to source the appropriate completion script.

**Pull request merged**